packages/core/src/fhirpath/parse.ts

```diff
@@ -259,5 +259,6 @@ export function evalFhirPathTyped(
   variables?: Record<string, TypedValue>
 ): TypedValue[] {
   const variableInput = variables ? variables : {};
-  return parseFhirPath(expression).eval({ variables: variableInput }, input);
+  const ast = parseFhirPath(expression);
```

Comment:
Breaking this out onto its own line was helpful for debugging, to inspect the parsed tree of atoms before evaluating the expression